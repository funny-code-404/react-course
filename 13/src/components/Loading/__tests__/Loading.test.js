import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { Loading } from "..";

configure({ adapter: new Adapter() });

describe('MAIN LOADING', () => {

    const props = {
        civilizations: { isFetching: false },
        units: { isFetching: false },
        structures: { isFetching: false },
        technologies: { isFetching: false },
        item: { isFetching: false },
        createdIn: { isFetching: false },
        uniqueTech: { isFetching: false },
        uniqueUnit: { isFetching: false },
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Loading {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Component is defined', () => {
        expect(wrapper).toHaveLength(1)
    });
    it('component`s state should change', () => {
        wrapper.setProps({
            civilizations: { isFetching: true }
        })
        expect(wrapper.find('.loadin')).toHaveLength(1)
        expect(wrapper.find('.loader')).toHaveLength(1)
        expect(wrapper.find('.inner')).toHaveLength(3)
    })
});

// npm run test src/components/Loading