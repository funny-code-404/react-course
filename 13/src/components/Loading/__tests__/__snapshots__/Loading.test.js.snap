// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MAIN LOADING to Match Snapshot 1`] = `""`;
